Algorithms/Backtracking/N Queen Problem.cpp

```diff
@@ -0,0 +1,113 @@
+#define N 4 
+#include <stdbool.h> 
+#include <stdio.h> 
+
+/* A utility function to print solution */
+void printSolution(int board[N][N]) 
+{ 
+	for (int i = 0; i < N; i++) { 
+		for (int j = 0; j < N; j++) 
+			printf(" %d ", board[i][j]); 
+		printf("\n"); 
+	} 
+} 
+
+/* A utility function to check if a queen can 
+be placed on board[row][col]. Note that this 
+function is called when "col" queens are 
+already placed in columns from 0 to col -1. 
```

Comment:
Paste it on top with name of algorithm and description

Reply:
@Bhavnaharitsa please review.
Any further changes required?

Reply:
Do the changes mentioned below